cosmo_tester/test_suites/image_based_tests/agent_installation_test.py

```diff
@@ -199,41 +82,69 @@ def test_ubuntu_trusty_provided_userdata_agent(cfy,
                                                tmpdir,
                                                logger):
     name = 'cloudify_agent'
-    user = attributes.ubuntu_username
+    os_name = 'ubuntu_14_04'
```

Comment:
We don't have this in the attributes as well?

Reply:
No, I changed it to have a username per OS (the way we did for centos and rhel even though they were the same username). This allowed me to DRY the code a lot more by having them all use the same set of inputs- e.g.
        'user': attributes['{os}_username'.format(os=os_name)],

Reply:
👍